workflow-templates/library-checks-and-release.yml

```diff
@@ -0,0 +1,57 @@
+name: Library checks & release
+
+on:
+  push:
+    branches:
+      - master
+      - next
+      - alpha
+      - beta
+
+env:
+  GH_TOKEN: ${{ secrets.GH_TOKEN }}
+  PULL_REQUEST_NUMBER: ${{ github.event.pull_request.number }}
+  SURGE_LOGIN: ${{ secrets.SURGE_LOGIN }}
+  SURGE_TOKEN: ${{ secrets.SURGE_TOKEN }}
+
+jobs:
+  main:
+    name: Run checks and deploy
+    runs-on: [self-hosted, fear]
+
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v2
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+
+      - name: Set GitHub packages registry
+        run: |
+          npm config set '//npm.pkg.github.com/:_authToken' ${{ secrets.GH_TOKEN }}
+          npm config set @typeform:registry https://npm.pkg.github.com/
+
+      - name: Get yarn cache
+        uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}-${{ hashFiles('.github/workflows/**.yml') }}
+
+      - name: Install Node.js dependencies
+        if: steps.yarn-cache.outputs.cache-hit != 'true'
+        run: yarn install --frozen-lockfile
+
+      - name: Run linting and tests
+        run: npx npm-run-all -p test lint
+
+      - name: Build assets
+        run: yarn build
+
+      - name: Release master
+        run: yarn semantic-release
+
+      - name: Deploy master
+        run: yarn deploy
```

Comment:
I'd remove this step. The `deploy` bit only makes sense if you have a storybook/assets you deploy later on too, after publishing a version (so, not most of the cases of a library)

Reply:
You can remove this too from above then, correct? 
```
      - name: Build assets
        run: yarn build
```

Reply:
Nope. This is used to actually build the library that will ultimately get published.